OutOfSchool/OutOfSchool.WebApi/Controllers/TeacherController.cs

```diff
@@ -101,12 +113,15 @@ public async Task<ActionResult> Update(TeacherDTO teacherDto)
         /// <param name="id">Teacher's key.</param>
         /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
         [Authorize(Roles = "organization,admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         [HttpDelete("{id}")]
-        public async Task<ActionResult> Delete(long id)
+        public async Task<IActionResult> DeleteTeacher(long id)
         {
-            if (id == 0)
+            if (id < 1 || teacherService.GetAll().Result.AsQueryable().Count() < id)
```

Comment:
teacherService.GetAll()
we retrieve all data from the db just to check the count
this is very unproductive, please change